src/components/spaces/show/index.js

```diff
@@ -170,7 +185,11 @@ export default class SpacesShow extends Component {
                 actionState={space.canvasState.actionState}
                 canBePrivate={canBePrivate}
                 onPublicSelect={this.onPublicSelect.bind(this)}
-                onPrivateSelect={this.onPrivateSelect.bind(this)}
+                onPrivateSelecundot={this.onPrivateSelect.bind(this)}
+                onUndo={this.onUndo.bind(this)}
+                onRedo={this.onRedo.bind(this)}
+                canUndo={space.checkpointMetadata.head !== space.checkpointMetadata.length - 1}
```

Comment:
Your null state in the selector (head: 0, length: 0) sets canUndo to true. Maybe null state should be head: 0 length: 1?